app/scripts/extensions/nav/dropdownMobile.js

```diff
@@ -1,10 +1,27 @@
 'use strict';
 
 angular.module('openshiftConsole')
-  .run(function(extensionRegistry) {
+  .run(function(extensionRegistry, Constants) {
     extensionRegistry
       .add('nav-dropdown-mobile', _.spread(function(user) {
-        return [{
+        var dropdownItems = [];
+
+        var launcherApps = Constants.APP_LAUNCHER_NAVIGATION;
+        _.each(launcherApps, function(app) {
+          var liObj = {
+            type: 'dom',
+            node: [
+              '<li>',
+              '<a href="' + _.escape(app.href) + '">',
+              '<span class="' + _.escape(app.iconClass) + '" aria-hidden="true"></span> ' + _.escape(app.title),
```

Comment:
Filed issue https://github.com/patternfly/patternfly/issues/648  and will follow up with @cdcabrera .